src/modules/hip/kernel/fog.cpp

```diff
@@ -1,90 +1,103 @@
 #include <hip/hip_runtime.h>
-#define saturate_8u(value) ( (value) > 255 ? 255 : ((value) < 0 ? 0 : (value) ))
-extern "C" __global__ void fog_planar(   unsigned char* input,
-                    const unsigned int height,
-                    const unsigned int width,
-                    const unsigned int channel,
-                    const float fogValue
-)
+
+#if defined(STATIC)
+#include "rpp_hip_host_decls.hpp"
+#endif
+
+#define saturate_8u(value) ((value) > 255 ? 255 : ((value) < 0 ? 0 : (value)))
+
+extern "C" __global__ void fog_planar(unsigned char *input,
+                                      const unsigned int height,
+                                      const unsigned int width,
+                                      const unsigned int channel,
+                                      const float fogValue)
 {
-    int id_x = hipBlockIdx_x *hipBlockDim_x + hipThreadIdx_x;
-    int id_y = hipBlockIdx_y *hipBlockDim_y + hipThreadIdx_y;
-    int id_z = hipBlockIdx_z *hipBlockDim_z + hipThreadIdx_z;
-    if (id_x >= width || id_y >= height) return;
-    int pixId= width * id_y  + id_x;
-    int c=width*height;
-    float check=input[pixId];
-    if(channel>1)
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x >= width || id_y >= height)
     {
-        check+=input[pixId+c]+input[pixId+c*2];
-        check=check/3;
+        return;
+    }
+
+    int pixId= width * id_y + id_x;
+    int c = width * height;
+    float check = input[pixId];
+    if(channel > 1)
+    {
+        check += input[pixId+c] + input[pixId + c * 2];
+        check = check / 3;
     }
     if(check >= (240) && fogValue!=0)
     {}
     else if(check>=(170))
     {
-        float pixel = ((float) input[pixId])  * (1.5 + fogValue) - (fogValue*4) + (7*fogValue);
+        float pixel = ((float) input[pixId]) * (1.5 + fogValue) - (fogValue*4) + (7 * fogValue);
```

Comment:
need to compute all 3 values at one time using vector instructions

Reply:
We'll do the kernel level optimizations subsequent PRs